### DISCUSSION TITLE
ZB-GW03 Zigbee Gareway Firmware Version

### DISCUSSION THREAD

**Discussion by jonny190** [arendst/Tasmota#21361] [Q&A]
Has anyone got any newer firmware then 6.7.8.0 build 373 or know how to build newer? as the lateset zigbee2mqtt is no longer supported being less then version 8 

`warning:  zh:ezsp: Deprecated driver 'ezsp' currently in use, 'ember' will become the officially supported EmberZNet driver in next release. If using Zigbee2MQTT see https://github.com/Koenkk/zigbee2mqtt/discussions/21462`

when switching to ember

```
[2024-05-05 14:53:20] error:    z2m: Error: NCP EZSP protocol version of 8 does not match Host version 13
    at EmberAdapter.emberVersion (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:1651:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EmberAdapter.initEzsp (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:802:9)
    at EmberAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:2660:24)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:124:29)
    at Zigbee.start (/app/lib/zigbee.ts:62:27)
    at Controller.start (/app/lib/controller.ts:108:27)
    at start (/app/index.js:107:5)

```

**Comment by jonny190** [comment]
Hardware for ref is https://templates.blakadder.com/ewelink_ZB-GW03.html

**Comment by sfromis** [comment]
I'm having 6.7.9 installed, as supplied by the Tasmota project:
https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp
Not at all "new", but a higher number than 6.7.8.

In practice, my Zigbee setup is primarily using a bridge with CC2652 instead.

**Reply by jonny190** [reply]
Just tried this but unfurtunatly the same error in Zigbee2MQTT :(

**Reply by sfromis** [reply]
This repository includes signed files with version 6.7.10 and 6.10.9.
https://github.com/xsp1989/zigbeeFirmware
I'm not sure if those are directly compatible with the built-in flashing setup of Tasmota. And while they are not tested with Tasmota, this is less important if you are going to use Z2M anyway.

**Reply by jonny190** [reply]
Slightly newer but still giving me 
```
[2024-05-05 18:17:44] error:    z2m: Error: NCP EZSP protocol version of 8 does not match Host version 13
    at EmberAdapter.emberVersion (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:1651:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EmberAdapter.initEzsp (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:802:9)
    at EmberAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:2660:24)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:124:29)
    at Zigbee.start (/app/lib/zigbee.ts:62:27)
    at Controller.start (/app/lib/controller.ts:108:27)
    at start (/app/index.js:107:5)
Using '/app/data' as data directory
```

Looks like i might need a new adaptor :(

**Reply by s-hadinger** [reply]
Zigbee2Tasmota supports only EZSP v8 protocol. Anyways CC2652P is largely taking over.

**Comment by jonny190** [comment]
> Zigbee2Tasmota supports only EZSP v8 protocol. Anyways CC2652P is largely taking over.


This shouldnt matter if its beeing used to just  export poty 8888 for Zigbee2MQTT to connect to? I could be wrong i'm just figuring this out as i go along

**Reply by s-hadinger** [reply]
Yes I know. It's just to indicate that we don't have any interest for the v13 in Tasmota currently

**Comment by robvanoostenrijk** [comment]
Has anyone succesfully flashed an updated newer Zigbee firmware then ncp-uart-sw_6.7.10_115200.ota on the ZB-GW03?

From what I understand the zbdonglee Zigbee firmware should be compatible *if* the ZB-GW03 accepts unsigned firmware:
https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee

However, I have no way to recover from a bad flash. Did anyone test if the ZB-GW03 can support 7x series Zigbee firmware?

**Reply by sfromis** [reply]
> should be compatible *if* the ZB-GW03 accepts unsigned firmware

It does not, if you use the standard serial upload process in Tasmota. I suppose that it would be possible if you instead connect directly to the Zigbee module. Not having worked with this, I don't have details about the steps for that.

**Comment by Hedda** [comment]
> > should be compatible _if_ the ZB-GW03 accepts unsigned firmware
> 
> It does not

Suggest try post feature request to @xsp1989 as he released the previous signed firmware -> https://github.com/xsp1989/zigbeeFirmware

**Comment by Hedda** [comment]
For reference, we can speculate that the open-source [Zigbee2MQTT (a.k.a. Z2M)](https://www.zigbee2mqtt.io/) Zigbee Gateway software application will soon probably require EZSP v13 (EmberZNet 7.4.0.0 firmware) or later as their minimum supported version because they just now marked their old `ezsp` adapter code (Silicon Labs EmberZNet Zigbee radio hardware-abstraction adapter code) as deprecated in zigbee-herdsman:

* https://github.com/Koenkk/zigbee-herdsman/pull/1039

`Deprecated driver 'ezsp' currently in use, 'ember' will become the officially supported EmberZNet ` +  `driver in next release. If using Zigbee2MQTT see https://github.com/Koenkk/zigbee2mqtt/discussions/21462`

So to clearify, Z2M has not yet dropped support for their `ezsp` adapter but them marking as deprecated is an indication they might do so.

This may still be of interest to other Zigbee projects, like Tasmota, that can also be used Silicon Labs Zigbee Coordinator adapters because of the fact that Zigbee2MQTT's new `ember` adapter code will only be compatible with EZSP v13 (EmberZNet 7.4.0.0 firmware) and later. In addition, RCP Multi-PAN/Multi-protocol firmware will also not be officially supported, instead only NCP firmware will be supported, at least for now. 

* https://thehelpfulidiot.com/a-wired-sonoff-zigbee-alternative
  * https://thehelpfulidiot.com/update-2-a-wired-sonoff-zigbee-alternative

zigbee-herdsman's new `ember` adapter code is meant to fully replace old `ezsp` adapter code but it does sacrifice backwards compatibility.

Zigbee2MQTT's developers on purpose choose to do a clean rewrite their new `ember` adapter code as "greenfield project" with focus on only supporting EZSP v13 (EmberZNet 7.4.0.0) firmware and later to avoid constraints and limitations imposed by old code or old firmware.

* https://github.com/Koenkk/zigbee2mqtt/discussions/21462
* https://github.com/Koenkk/zigbee-herdsman/pull/918

(Maybe it is worth having a higher minimum version of EZSP as supported to avoid a possible worse user-experience with older versions?)

Anyway, deprecated in ZigbeeMQTT will initially only mean that their old `ezsp` adapter code will no longer maintained and that users will have an yet undefined grace period to migrate to the new `ember `adapter.  Silicon Labs adapters are currently still "experimental" in Z2M -> https://www.zigbee2mqtt.io/guide/adapters/#experimental however their plan is that this will change in the future with their new `ember` adapter code becomes more stable, as then they might completely remove their old `ezsp` adapter code.

* https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ember
* https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ezsp

Note that Zigbee2MQTT has previously indirectly also made old CC235x adapter code deprecated by listing that as not recommended:

* https://www.zigbee2mqtt.io/guide/adapters/

PS: I am not sure how well backup and restore is working today if using their old `ezsp` adapter in Zigbee2MQTT but at least until recently there was no ability to restore from backup in Z2M if using that `ezsp` adapter in Zigbee2MQTT, see -> https://github.com/Koenkk/zigbee-herdsman/issues/319 and the workaround to backup and restore before if using `ezsp` adapter in Zigbee2MQTT was to backup and restore using zigpy-cli, see https://github.com/Koenkk/zigbee2mqtt/discussions/16478 but I believe the new `ember` adapter in Zigbee2MQTT has full native support for backup and restore but I am not sure about native migrations from `ezsp` adapter to `ember` adapter inside Zigbee2MQTT.

**Comment by Mik145** [comment]
Hi all!
Zigbee2Mqtt has been updated to version 1.38.0-1. 
Will my ZB-GW03 gateway with firmware version EZSP v8 6.7.10.0 build 423 work correctly in this version?

**Reply by Hedda** [reply]
> Zigbee2Mqtt has been updated to version 1.38.0-1.
> Will my ZB-GW03 gateway with firmware version EZSP v8 6.7.10.0 build 423 work correctly in this version?

Not if you want to migrate to the new ”Ember” adapter in Zigbee2MQTT that I menationed above, as then you first need to upgrade to the EmberZNet 7.4.x.x firmware, see experimental signed build here:

https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/ZigbeeBridge_SM-011-signed

More discussion about that specifc firmware build here:

* https://github.com/xsp1989/zigbeeFirmware/issues/49#issuecomment-2137949160

As I understand, you need to have upgraded to at least EmberZNet 7.4.x firmware version or later to use the new ”ember” driver/adapter in Zigbee2MQTT, but from what I read it is suggested to first upgrade to EmberZNet 7.3.x firmware version while still using the old ”ezsp” driver/adapter as it sounds like a migration to the new ”ember” driver will go smoother if already have had the old ”ezsp” driver/adapter working EmberZNet 7.3.x firmware version while still using the old ”ezsp” driver/adapter in Zigbee2MQTT.

Again, recommend read and ask more specific questions in Zigbee2MQTT’s own community forum on GitHub after you done more research. The multi-step process you need to do depends on your specific setup and you need to ask in the Zigbee2MQTT community if you need step-by-step that applies to your setup. The process for actually updating firmware version will not be different.

Suggest start by reading and try following this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:

- https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/

Summery; `ezsp` adapter is now deprecated in Zigbee2MQTT’s zigbee-herdsman library, which means that the Zigbee2MQTT developers are by doing so strongly recommending users to upgrade and migrate to `ember` adapter, and that migration might require a multi-step process depending on what EmberZNet firmware version you are using today. Users that continues to use the `ezsp` adapter will for now simply get a warning in their logs that they are using a deprecated adapter, however it will not stop working.

Note! If you have a EFR32MG12/MGM12 series, EM35x/EM35xx or ETRX35x based adapter then you need to buy a newer adapter. If you want to migrate to newer Silicon Labs based Zigbee Coordinator today then you should consider getting an EFR32MG2x (EFR32MG21 or EFR32MG24) based USB radio dongle, (though technically you can even upgrade EFR32MG13):

* https://www.zigbee2mqtt.io/guide/adapters/
  * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

For reference see:

* https://github.com/Koenkk/zigbee-herdsman/pull/918
* https://github.com/Koenkk/zigbee2mqtt/discussions/21462
  * https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320
* https://github.com/Koenkk/zigbee-herdsman/pull/1039


PS: "Deprecated" does not mean that the `ezsp` adapter will be removed yet, instead it means you are discouraged from using it as it is being phased out and replaced by the newer `ember` adapter (which is not compatible with older firmware or all hardware).

* https://en.wikipedia.org/wiki/Deprecation#Software
* https://techterms.com/definition/deprecated
* https://www.baeldung.com/cs/deprecated-vs-obsolete

**Reply by Mik145** [reply]
> > Zigbee2Mqtt has been updated to version 1.38.0-1.
> > Will my ZB-GW03 gateway with firmware version EZSP v8 6.7.10.0 build 423 work correctly in this version?
> 
> Not if you want to migrate to the new ”Ember” adapter in Zigbee2MQTT that I menationed above, as then you first need to upgrade to the EmberZNet 7.4.x.x firmware, see experimental signed build here:
> 
> https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/ZigbeeBridge_SM-011-signed
> 
> More discussion here:
> 
> * [EmberZNet 7.3.2.0/7.4.0.0 firmware xsp1989/zigbeeFirmware#49 (comment)](https://github.com/xsp1989/zigbeeFirmware/issues/49#issuecomment-2137949160)

Thanks for the help and links with firmware!
With a little dancing with a tambourine, I updated my ZB-GW03 to the firmware "ncp-uart-sw_7.4.3.0_115200". It’s time to save the gateway settings, reset the settings to factory settings, flash v6.10.9, v7.3.1.0, v7.4.3.0 one by one and restore your settings.
Now I'm using the Ember driver.

**Reply by raedkit** [reply]
@Mik145 do you have any article or notes on how to this ? I think this could be helpful.
Thanks in advance

**Reply by Hedda** [reply]
See comment by @mvadu below:

> I followed this [xsp1989/zigbeeFirmware#49 (comment)](https://github.com/xsp1989/zigbeeFirmware/issues/49#issuecomment-2137189283) and was able to switch to ember. Inside Z2M had to change the protocol to ember too.
> 
> I was able to upgrade to ember firmware without having to repair any of the devices. Using https://github.com/xsp1989/ZB-GW03_firmware/tree/master/tasmota/v14.0.0/firmware as main tasmota firmware.

Recommended also read and follow this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:

https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/

Summery; `ezsp` adapter is now deprecated which means it is strongly recommended to upgrade and migrate to `ember` adapter, and take care that migration might require a multi-step process depending on what EmberZNet firmware version you use today. Users that continues to use the `ezsp` adapter will get a warning in their logs that they are using a deprecated adapter.

Note! If you have a EFR32MG12/MGM12 series, EM35x/EM35xx or ETRX35x based adapter then you need to buy a newer adapter. If you want to migrate to newer Silicon Labs based Zigbee Coordinator today then you should consider getting an EFR32MG2x (EFR32MG21 or EFR32MG24) based USB radio dongle, (though technically you can even upgrade EFR32MG13):

* https://www.zigbee2mqtt.io/guide/adapters/
  * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

For reference see:

* https://github.com/Koenkk/zigbee-herdsman/pull/918
* https://github.com/Koenkk/zigbee2mqtt/discussions/21462
  * https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320
* https://github.com/Koenkk/zigbee-herdsman/pull/1039


PS: "Deprecated" does not mean that the `ezsp` adapter will be removed yet, instead it means you are discouraged from using it as it is being phased out and replaced by the newer `ember` adapter (which is not compatible with older firmware or all hardware).

* https://en.wikipedia.org/wiki/Deprecation#Software
* https://techterms.com/definition/deprecated
* https://www.baeldung.com/cs/deprecated-vs-obsolete

**Comment by basilfx** [comment]
I built several 7.x images by [modifying](https://github.com/basilfx/silabs-firmware-builder) Nabu Casa's firmware builder to also build for the ZB-GW03. I've replaced the bootloader (using JTAG) and now I run (unsigned) images as close as possible to how Nabu Casa builds them for ZHA.

Latest images can be found here: https://github.com/basilfx/silabs-firmware-builder/actions/runs/9652452045#artifacts

Note that the image is called `xxx-ncp-uart-hw_7.4.2.0`, but it does not use flow control. It's just naming dictated by Nabu Casa's build tools.

**Reply by Hedda** [reply]
Maybe should clearify that can not use unsigned firmware for ZB-GW03 unless have also first manually flashed new bootloader to the Silicon Labs EFR32MG21 based Zigbee radio module with a compatible JTAG debug probe and what that means.

That is, unless you flashed new bootloader to the Silicon Labs EFR32MG21 (which most users do not do) then you are going to use a a signed firmware from xsp1989 if and when he build those for the SM-011 module inside these (as he is the only one with the signature needed to sign SM-011 module for the default bootloader that they ship with):

https://github.com/xsp1989/zigbeeFirmware/

**Reply by mvadu** [reply]
I followed this https://github.com/xsp1989/zigbeeFirmware/issues/49#issuecomment-2137189283 and was able to switch to ember. Inside Z2M had to change the protocol to ember too.

I was able to upgrade to ember firmware without having to repair any of the devices. Using https://github.com/xsp1989/ZB-GW03_firmware/tree/master/tasmota/v14.0.0/firmware as main tasmota firmware.

**Reply by Mik145** [reply]
> Maybe should clearify that can not use unsigned firmware for ZB-GW03 unless have also first manually flashed new bootloader to the Silicon Labs EFR32MG21 based Zigbee radio module with a compatible JTAG debug probe and what that means.
> 
> That is, unless you flashed new bootloader to the Silicon Labs EFR32MG21 (which most users do not do) then you are going to use a a signed firmware from xsp1989 if and when he build those for the SM-011 module inside these (as he is the only one with the signature needed to sign SM-011 module for the default bootloader that they ship with):

Hi. Do you know how to get away from the signed firmware on the ZB-GW03? I have a backup gateway, it would be interesting to try

**Reply by s-hadinger** [reply]
It's basically designed so that you can't flash a non-signed firmware.

If you really want, you can use a JTAG probe (very few support EFR32MG21), and you can completely erase the flash which removes the bootloader. Then you can flash back a non-signed bootloader and firmware.

I did this once with a Segger once, but it's not easy and not straightforward. Also the cost of a Segger probe is far too expensive compared to the price of the device. There was a tentative to make it workable with an opensource solution, but I doubt it went anywhere.

Long story short, the only way I know is to use a Segger JTAG (actually SWD) probe

**Reply by Mik145** [reply]
Thanks for the answer.
I looked at the price of Segger JTAG. You are right! It is much easier to buy another gateway)))). Now, ZB-GW03 works fine with ncp-uart-sw_7.4.3.0_115200. I asked more out of interest than because of the need to update)

**Reply by tyjtyj** [reply]
> uart-sw_7.4.3.0_115200. I asked more out of interest than because of the need to update)

any guide how to upgrade to 7.4 from tasmota ?

**Reply by sfromis** [reply]
Same way as when flashing your current firmware version.
https://templates.blakadder.com/ewelink_ZB-GW03.html

**Reply by xsp1989** [reply]
> It's basically designed so that you can't flash a non-signed firmware.
> 
> If you really want, you can use a JTAG probe (very few support EFR32MG21), and you can completely erase the flash which removes the bootloader. Then you can flash back a non-signed bootloader and firmware.
> 
> I did this once with a Segger once, but it's not easy and not straightforward. Also the cost of a Segger probe is far too expensive compared to the price of the device. There was a tentative to make it workable with an opensource solution, but I doubt it went anywhere.
> 
> Long story short, the only way I know is to use a Segger JTAG (actually SWD) probe

I'll correct you. Most jlink_v9s should be able to be de-flashed and re-burned. Using the command line, you can easily re-flash the firmware using the command line. The price of jlinkv9 in China is about 70-90 RMB, which is not expensive.
If there are a lot of users, at least more than 100 users will go to my GitHub request, I can regenerate another pair of keys and release a firmware that can be used to update the public key in SM011/ZYZBP008. In this way, you can burn the firmware freely without re-burning the bootloader, but you still need to sign it with another private key I released (this private key I will make public). The mandatory signature feature is a requirement of the bootloader and cannot be updated.
Warning: Of course, after the update, you will no longer be able to use the signed firmware I released for SM-011/ZYZBP008, but you can sign the unsigned ZB-GW04 firmware I released to update your gateway, and I will no longer release any firmware for this public key pair.

**Reply by s-hadinger** [reply]
Thanks for the clarification that JLink v9 works.

**Reply by Mik145** [reply]
> > uart-sw_7.4.3.0_115200. I asked more out of interest than because of the need to update)
> 
> any guide how to upgrade to 7.4 from tasmota ?

I just reset the gateway settings, updated all the firmware in turn through the Tasmota web interface. Reconfigured the gateway

**Comment by Mik145** [comment]
my coordinator ZB-GW03 Zigbee version 7.4.3.0 build 0 (Using "Ember" in z2m settings)
After updating z2m to v1.40.1, an error regularly appears:

```
Logger: bellows.types.named
Source: components/zha/__init__.py:151
First occurred: 1:26:04 PM (4 occurrences)
Last logged: 1:26:07 PM

Unknown status <EmberStatus.LIBRARY_NOT_PRESENT: 181>, converting to generic <sl_Status.FAIL: 1>
```
What could it be?
Google does not give an answer(

**Reply by s-hadinger** [reply]
This is a question to ask to the ZHA community. We have no clue what this code does...

**Reply by Mik145** [reply]
Thanks for the answer! I completely forgot that I recently launched another ZB-GW03 on ZHA))

**Reply by Hedda** [reply]
> After updating z2m to v1.40.1, an error regularly appears:

To clearify, it should be asked to he Zigbee2MQTT community if using Z2M -> https://github.com/Koenkk/zigbee2mqtt/discussions

**Reply by Mik145** [reply]
> o clearify, it should be asked to he Zigbee2MQTT community if using Z2M -> https://github.com/Koenkk/zigbee2mqtt/discussions

I have two ZB-GW03 gateways with EmberZNet 7.4.3 [GA] firmware. One gateway works on Z2M, the other on ZHA. The error occurs on ZHA. There are no problems on z2m. Apparently ZHA does not yet support EmberZNet 7.4.3 [GA] firmware. Is it possible to downgrade zigbee firmware on ZB-GW03?

**Reply by Hedda** [reply]
> I have two ZB-GW03 gateways with EmberZNet 7.4.3 [GA] firmware. One gateway works on Z2M, the other on ZHA. The error occurs on ZHA. There are no problems on z2m. Apparently ZHA does not yet support EmberZNet 7.4.3 [GA] firmware. Is it possible to downgrade zigbee firmware on ZB-GW03?

ZHA does support all EmberZNet 7.4.x.x firmware (only EmberZNet 8.0.x.x firmware is experimental in ZHA) so you should just open an a issue for ZHA to Home Assistent core repository if there is not already one -> https://github.com/home-assistant/core/issues?q=is%3Aissue+zha (however be sure to first read and follow this -> https://www.home-assistant.io/integrations/zha#reporting-issues and also this -> https://www.home-assistant.io/integrations/zha#debug-logging )

**Comment by speedycars** [comment]
What about 7.4.4? Seems like some cool fixes were implemented in it.

**Reply by Hedda** [reply]
Post a feature request for signed firmware as new issue to if is not anready one posted -> https://github.com/xsp1989/zigbeeFirmware

**Comment by AldoLodos** [comment]
Hello, 
just bought a ZB-GW03 gateway
can somebody tell me where to download teh EmberZNet 7.4.3 [GA] firmware ?
Thanks

**Comment by AldoLodos** [comment]
Hello again,
I found and flashed a 7.4.3  firmware in my ZB-GW03. It was successful but the console shows an error, any idea ?

00:00:00.002 HDW: ESP32-D0WD-V3 v3.1 
00:00:00.040 UFS: FlashFS mounted with 304 kB free
00:00:00.062 CFG: Loaded from File, Count 9
00:00:00.070 QPC: Count 1
00:00:00.075 I2C: Bus1 using GPIO32(SCL) and GPIO33(SDA)
00:00:00.149 BRY: Berry initialized, RAM used=3902 bytes
00:00:00.164 TFS: File '.drvset003' not found
00:00:00.166 ZIG: ZBBridge EEPROM found at address 0x50
00:00:00.173 Project tasmota - Tasmota Version 13.2.0(tasmota)-2_0_14(2023-10-20T06:21:02)
00:00:01.641 ETH: Connected at 100Mbps Full Duplex, Mac EC:64:C9:81:D0:67, Hostname tasmota-81D064-4196-eth
00:00:01.690 ZIG: Resetting EZSP device
01:03:22.030 WIF: WifiManager active for 3 minutes
01:03:22.745 HTP: Web server active on tasmota-81D064-4196-eth with IP address 192.168.2.47
01:03:22.758 mDN: Initialized 'tasmota-81D064-4196.local'
01:03:22.759 HTP: Web server active on tasmota-81D064-4196-eth.local with IP address 192.168.2.47
01:03:23.804 RSL: INFO1 = {"Info1":{"Module":"ZB-GW03","Version":"13.2.0(tasmota)","FallbackTopic":"cmnd/DVES_81D064_fb/","GroupTopic":"cmnd/tasmotas/"}}
01:03:23.817 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"tasmota-81D064-4196-eth","IPAddress":"192.168.2.47"}}}
01:03:23.829 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":3}}
01:03:27.961 QPC: Reset
01:03:27.965 RSL: STATE = {"Time":"2024-09-29T01:03:27","Uptime":"0T00:00:06","UptimeSec":6,"Heap":146,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":38}}
01:03:37.208 ZIG: Resetting EZSP device
01:03:38.460 RSL: RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
01:03:38.560 RSL: RESULT = {"ZbState":{"Status":55,"Version":"7.4.3.0","Protocol":13,"Stack":2}}
01:03:38.563 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
01:03:38.576 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
01:03:38.577 ZIG: Abort
01:03:38.578 ZIG: Stopping (99)
01:08:27.958 RSL: STATE = {"Time":"2024-09-29T01:08:27","Uptime":"0T00:05:06","UptimeSec":306,"Heap":142,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":38}}

**Reply by sfromis** [reply]
If you want to use it with Z2T, you should limit yourself to supported versions, instead of versions no longer compatible.

If you want to use it with Z2M or ZHA, you could configure for a TCP tunnel letting such projects get full control over the Zigbee chip.

**Reply by AldoLodos** [reply]
Thanks for your answer,
I want to use it with Home Assistant and Z2M. Is there examples of TCP tunnel config ?

However I 'm confusing about the message in the console 

01:03:38.563 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
01:03:38.576 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
01:03:38.577 ZIG: Abort
01:03:38.578 ZIG: Stopping (99)

Does it mean that I don't have the right firmware ?

**Reply by sfromis** [reply]
The basics is that you need to reconfigure the Zigbee Tx/Rx pins to TCP Tx/Rx instead, and issue a `TCPStart` command (with port number) after boot. Of course also configure Z2M for such a connection.

After this config, Tasmota is totally out of the loop when it comes to Zigbee, and the question about "right firmare" is up to what's supported by Z2M.

**Reply by Hedda** [reply]
@AldoLodos see these articles which covers all steps needed for use with ZHA and Zigbee2MQTT:

https://thehelpfulidiot.com/update-a-wired-sonoff-zigbee-alternative

https://templates.blakadder.com/ewelink_ZB-GW03.html

https://github.com/vahempio/Tasmota-for-eWeLink

and of course also use this for reference

https://tasmota.github.io/docs/Zigbee/#sonoff-snzb-02-sensor

**Reply by AldoLodos** [reply]
Hi guys,
thanks for your answers and links 
I succeded to setup my ZB_GW03 to operate with HA and Z2M with the ncp-uart-sw_7.4.3.0_115200.ota firmware
I really appreciated your help

**Reply by jsanfe** [reply]
Sorry @AldoLodos ,, how you did this??  Where did you found ncp-uart-sw_7.4.3.0_115200.ota?

**Reply by sfromis** [reply]
Probably from the project already mentioned several times in this thread:
https://github.com/xsp1989/zigbeeFirmware

**Comment by lucize** [comment]
> 01:03:38.563 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
01:03:38.576 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
01:03:38.577 ZIG: Abort
01:03:38.578 ZIG: Stopping (99)

for me, I had to upload the updated firmware from 7.x.x to other 7.x.x as soon as possible, not to let this message to appear otherwise nothing would happen on *.ota update

**Reply by sfromis** [reply]
Yeah, without the right protocol version, the only way would be TCP ser2net mode with Z2M or ZHA, where Tasmota is no longer doing anything Zigbee.

**Reply by xsp1989** [reply]
> > 01:03:38.563 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
> > 01:03:38.576 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
> > 01:03:38.577 ZIG: Abort
> > 01:03:38.578 ZIG: Stopping (99)
> 
> for me, I had to upload the updated firmware from 7.x.x to other 7.x.x as soon as possible, not to let this message to appear otherwise nothing would happen on *.ota update

I tried to get Tasmota to support firmware version 7.x, namely EZSP V13, but it's not finished yet.

**Comment by fhloston** [comment]
I can confirm that using simplicity commander and a j-link mini edu v2 the device can be reset and a new bootloader flashed.

Since i have two of these ZB-GW03 v1.3 gathering dust I am interested in converting them to the ot-rcp firmware.

I can successfully flash from unsigned bootloader to ot-rcp-v2.4.5.0-zb-gw04-1v2-230400.gbl.

However to my understanding the gw04-1v2 firmware is built with hw flow control which the ZB-GW03 does not have routed.

So I guess i should build my own version of the firmware without hw flow control - is this correct or has someone already done that?

**Reply by basilfx** [reply]
See https://github.com/arendst/Tasmota/discussions/21361#discussioncomment-9864968

I have created bootloader and firmware images that works for NCP. It might be a starting point for RCP.

**Reply by fhloston** [reply]
thank you, forked your repo https://github.com/fhloston/silabs-firmware-builder and built the ot-rcp firmware with 230400 and no flow control.

it actually works now, i have the homeassistant openthread border router running with this over the streamserver on the zb-gw03.
